ui/apps/platform/cypress/integration/collections/deploymentMatching.test.js

```diff
@@ -0,0 +1,183 @@
+import withAuth from '../../helpers/basicAuth';
+import { hasFeatureFlag } from '../../helpers/features';
+import {
+    assertDeploymentResultCountEquals,
+    assertDeploymentsAreMatched,
+    assertDeploymentsAreNotMatched,
+    tryDeleteCollection,
+    visitCollections,
+} from './Collections.helpers';
+import { collectionSelectors as selectors } from './Collections.selectors';
+
+const baseUrl = '/v1/collections';
+const autocompleteUrl = `${baseUrl}/autocomplete`;
+
+describe('Collection deployment matching', () => {
+    withAuth();
+
+    const sampleCollectionName = 'Stackrox sample deployments';
+    const withEmbeddedCollectionName = 'Contains embedded collections';
+
+    beforeEach(function beforeHook() {
+        if (!hasFeatureFlag('ROX_OBJECT_COLLECTIONS')) {
+            this.skip();
+        }
+        // Ignore autocomplete requests
+        // TODO Remove this once the feature is in
+        cy.intercept(autocompleteUrl, {});
+    });
+
+    // Clean up when the test suite exits
+    after(() => {
+        if (hasFeatureFlag('ROX_OBJECT_COLLECTIONS')) {
+            tryDeleteCollection(withEmbeddedCollectionName);
+            tryDeleteCollection(sampleCollectionName);
+        }
+    });
+
+    it('should preview deployments matching specified rules', () => {
+        // Cleanup from potential previous test runs
+        tryDeleteCollection(withEmbeddedCollectionName);
+        tryDeleteCollection(sampleCollectionName);
+
+        visitCollections();
+
+        cy.get('a:contains("Create collection")').click();
+        cy.get('input[name="name"]').type(sampleCollectionName);
+        cy.get('input[name="description"]').type('Matches some stackrox deployments');
+
+        cy.get('button:contains("All namespaces")').click();
+        cy.get('button:contains("Namespaces with names matching")').click();
+        cy.get('input[aria-label="Select value 1 of 1 for the namespace name"]').type('stackrox');
+        cy.get(`button:contains('stackrox')`).click();
+
+        // Test that Stackrox deployments are matched
+        assertDeploymentsAreMatched('central', 'central-db', 'collector', 'scanner', 'sensor');
```

Comment:
For my understanding, what is the relationship between helper functions:
* `assertDeploymentsAreMatched`
* `assertDeploymentResultCount`

Here only one is called, below both are called.
* Is that because here there might be other results than the StackRox deployments?
* Below, where the count is the number of deployments, what do you think about `deployments` array argument instead of arguments, so `deployments.length` is the expected count?

Reply:
Yep, this is to ensure that 'collector' and 'sensor' are matched and that there are no other matches. Switching to an array arg and using the length is a good idea 👍

Reply:
Oh, here is a delayed thought. How about 2 functions whose names distinguish:
* match deployments, but allow others (current function behavior)
* match deployments, but do not allow any others (current function plus length function)

That is, second function calls the first and also encapsulates the length assertion.

Reply:
The assertion about the button precedes the others, so its failure indicates a false assumption in test itself?

Reply:
Yeah the assertion that the "view more" doesn't exist could be a false assumption in the test itself if the developer tried to pass in 10+ deployments for matching. We could add a length check and an explicit `throw` to make it more clear as a follow up.